spring-cloud-dataflow-server/docker-compose-prometheus.yml

```diff
@@ -0,0 +1,99 @@
+    image: springcloud/spring-cloud-dataflow-server:${DATAFLOW_VERSION:?DATAFLOW_VERSION is not set!}
+    container_name: dataflow-server
+    ports:
+      - "9393:9393"
+    environment:
+      - spring.cloud.dataflow.applicationProperties.stream.spring.cloud.stream.kafka.binder.brokers=kafka:9092
+      - spring.cloud.dataflow.applicationProperties.stream.spring.cloud.stream.kafka.binder.zkNodes=zookeeper:2181
+      - spring.cloud.skipper.client.serverUri=http://skipper-server:7577/api
+      - spring.cloud.dataflow.applicationProperties.stream.management.metrics.export.prometheus.enabled=true
+      - spring.cloud.dataflow.applicationProperties.stream.spring.cloud.streamapp.security.enabled=false
+      - spring.cloud.dataflow.applicationProperties.stream.management.endpoints.web.exposure.include=prometheus,info,health
+      - spring.cloud.dataflow.grafana-info.url=http://localhost:3000
+    depends_on:
+      - kafka
+  app-import:
+    image: springcloud/openjdk:latest
+    depends_on:
+      - dataflow-server
+    command: >
+      /bin/sh -c "
+        while ! nc -z dataflow-server 9393;
+        do
+          sleep 1;
+        done;
+        wget -qO- 'http://dataflow-server:9393/apps' --post-data='uri=http://bit.ly/Einstein-GA-stream-applications-kafka-maven&force=true';
+        echo 'Stream apps imported'
+        wget -qO- 'http://dataflow-server:9393/apps' --post-data='uri=http://bit.ly/Dearborn-SR1-task-applications-maven&force=true';
+        echo 'Task apps imported'"
+  skipper-server:
+    image: springcloud/spring-cloud-skipper-server:${SKIPPER_VERSION:?SKIPPER_VERSION is not set!}
+    container_name: skipper
+    ports:
+      - "7577:7577"
+      - "9000-9010:9000-9010"
+      - "20000-20105:20000-20105"
+    environment:
+      - SPRING_CLOUD_SKIPPER_SERVER_PLATFORM_LOCAL_ACCOUNTS_DEFAULT_PORTRANGE_LOW=20000
+      - SPRING_CLOUD_SKIPPER_SERVER_PLATFORM_LOCAL_ACCOUNTS_DEFAULT_PORTRANGE_HIGH=20100
+
+  # Grafana is configured with the Prometheus datasource.
+  # Use `docker exec -it prometheus /bin/sh` to logging in the container
+  prometheus:
+    image: springcloud/spring-cloud-dataflow-prometheus-local:${DATAFLOW_VERSION:?DATAFLOW_VERSION is not set! Use 'export DATAFLOW_VERSION=local-server-image-tag'}
+    container_name: 'prometheus'
+    volumes:
+      - 'scdf-targets:/etc/prometheus/'
+    ports:
+      - '9090:9090'
+    depends_on:
+      - service-discovery
+
+  # Grafana SCDF Prometheus pre-built image:
+  grafana:
+    image: springcloud/spring-cloud-dataflow-grafana-prometheus:${DATAFLOW_VERSION:?DATAFLOW_VERSION is not set! Use 'export DATAFLOW_VERSION=local-server-image-tag'}
+    container_name: 'grafana'
+    ports:
+      - '3000:3000'
+    depends_on:
+      - prometheus
+
+  # The service-discovery container.
+  # Use `docker exec -it service-discovery /bin/sh` to logging the container
+  service-discovery:
+    image: springcloud/spring-cloud-dataflow-prometheus-service-discovery:0.0.3
```

Comment:
What's the reasoning to go with the `0.0.3` version here? Shouldn't we use the SCDF versioning for the same?

Reply:
At the moment the service-discovery project is not part of the SCDF build ecosystem and therefore is not possible to automatically generate service-discovery  aligned with the SCDF versions.

Reply:
ok, let's discuss on what would be the appropriate place for this project and have it versioned before merging this PR.